### ISSUE TITLE
Permissions page: The sticky table header hides permission scrolled to using URL fragments.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3414]
## Describe your issue or idea

I have come across this in the past before, but now again when working in #3339...

Each row in the permissions page (`/admin/config/people/permissions`) has an id with the name of the permission. This allows to provide links to specific permissions within that page. For example `/admin/config/people/permissions#edit-view-hidden-paths` will scroll that page to the "View hidden paths" permission. Although this works as expected, the sticky table header in that page overlaps the row, hiding it from the view.

### Steps to reproduce (if reporting a bug)

Append the following to your site's URL: `/admin/config/people/permissions#edit-view-hidden-paths`

### Actual behavior (if reporting a bug)

Page scrolls to the "View hidden paths" permission, but the permission row is hidden by the sticky table header.

### Expected behavior (if reporting a bug)

The page scrolls to the "View hidden paths" permission, and there is enough padding/margin so that the sticky table header does not overlap with the permission row.

---
PR by @klonos (CSS-only solution): ...

**Comment by klonos** [comment]
PR up for review. I have chosen to go with a CSS-only solution instead of having to use JavaScript (because I suck at it 😅)...

The CSS fix is fairly simple, using `:target` to add a 50px padding-top to the table cells of the permission row being targeted by the URL fragment.

Adding the padding to the table cell with the permission name/description, would work somewhat as expected when the permission name and description would wrap in 3 rows:

![screen shot 2018-12-09 at 1 17 08 am](https://user-images.githubusercontent.com/2423362/49696639-9eff8b80-fbb5-11e8-931e-863000f71f9d.png)

...but when wrapping in 2 rows, or if there was no permission description, the checkboxes would be misaligned or still hidden behind the sticky table header:

![screen shot 2018-12-09 at 1 17 35 am](https://user-images.githubusercontent.com/2423362/49696646-bfc7e100-fbb5-11e8-9332-f98dd0f0be8a.png)

![screen shot 2018-12-09 at 1 17 48 am](https://user-images.githubusercontent.com/2423362/49696662-f271d980-fbb5-11e8-9f0a-742b2347b897.png)

In order to achieve that in a cleaner way, I had to move the permission IDs from the table cells with the permission name/description to the parent table row instead. So HTML code before:

```html
<table class="permissions" ...>
  <tbody>
    <tr class="odd">
      <td class="module" id="module-admin_bar" colspan="4">Administration bar</td>
    </tr>
    <tr class="even">
      <td class="permission">
        <div id="edit-access-administration-bar" class="form-item form-type-item">
          Access administration bar
          <div class="description"> ... </div>
        </div>
      </td>
      <td class="checkbox"> ... </td>
      <td class="checkbox"> ... </td>
      <td class="checkbox"> ... </td>
    </tr>
```

...and after:

```html
<table class="permissions" ...>
  <tbody>
    <tr id="module-admin_bar" class="odd">
      <td class="module" colspan="4">Administration bar</td>
    </tr>
    <tr id="edit-access-administration-bar" class="even">
      <td class="permission">
        <div class="form-item form-type-item">
          Access administration bar
          <div class="description"> ... </div>
        </div>
      </td>
      <td class="checkbox"> ... </td>
      <td class="checkbox"> ... </td>
      <td class="checkbox"> ... </td>
    </tr>
```

If you log into the PR sandbox, and then navigate to a specific permission in the permissions page, like http://2413.backdrop.backdrop.qa.backdropcms.org/admin/config/people/permissions#edit-administer-comments for example, you will be auto-scrolled to the "Administer comments and comment settings" permission, and it will be visible. The downside is that if you scroll up, you can see the padding that's been added:

![screen shot 2018-12-09 at 1 36 00 pm](https://user-images.githubusercontent.com/2423362/49696791-ef77e880-fbb7-11e8-8986-85fa03b0addf.png)

...the only way to get rid of that is to use JavaScript instead of a plain CSS solution. If anyone is up for it, be my guest 😄

**Comment by klonos** [comment]
Thanks for the review @amoutiers 👍 ...you are faster than me 😄

In the meantime, I have posted an explanation of the solution I have chosen to go with ^^

**Comment by quicksketch** [comment]
@klonos A common solution I've seen for this problem is to add the padding like you've done and move the selector onto a `<a name="foo">` tag that serves no purpose other than to exist on the page. Position it with a positive padding but a negative margin so it just moves upward from it's normal position, and then make it `visibility: hidden` and/or zero pixels tall.

**Comment by quicksketch** [comment]
I also think a JS fix might be appropriate here. For example we could create a class or data attribute that we could place on named anchors like this to automatically provide offset-adjusted anchors, e.g. `<a name="foo" data-anchor-offset-adjust="true">`

We also have a probably mostly unknown pattern of adding a data attribute to elements that increase the offset of sticky elements from the top of the page. For example we have a mechanism to keep the sticky table header from overlapping the admin_bar at the top. Any element that affects the top margin is _supposed_ to include a `data-offset-top` data attribute. Then other elements can adjust. See `tableHeaderOffsetChangeHandler()` in tableheader.js for examples.

**Comment by klonos** [comment]
Yep, as I said, a JS solution would have most likely worked better in this situation, but:

> ...I have chosen to go with a CSS-only solution instead of having to use JavaScript (because I suck at it 😅)

Anyone JS-savvy care to take over?

**Comment by jenlampton** [comment]
This issue still needs more feedback, removing milestone, but adding milestone candidate tag.

**Comment by Jack-Chin** [comment]
Looks good to me.

**Comment by olafgrabienski** [comment]
I think we can close this issue as duplicate of https://github.com/backdrop/backdrop-issues/issues/5862 which has been fixed in Backdrop 1.34.3.
